jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js

```diff
@@ -9,51 +9,42 @@ licensing for the GUI.
 var module = module;
 
 /**
- * jujulib provides API access for microservices used by juju.
- *
- * jujulib provies access to the APIs for the Juju Environment
- * Manager (JEM), the juju charmstore, and the juju identity
- * manager (IdM).
+   jujulib provides API access for microservices used by juju.
+
+   jujulib provies access to the APIs for the Juju Environment
+   Manager (JEM), the juju charmstore, and the juju identity
+   manager (IdM).
  */
 (function (exports) {
   'use strict';
 
   /**
-   * Environment object for jujulib.
-   *
-   * Provides access to the JEM API.
-   */
-
-  /**
-   * Initializer
-   *
-   * @function environment
-   * @param url {String} The URL, including scheme and port, of the JEM instance.
-   * @param bakery {Object} A bakery object for communicating with the JEM instance.
-   * @returns {Object} A client object for making JEM API calls.
-   */
-  function environment(url, bakery) {
-    this.jemUrl = url + '/v1';
-    this.bakery = bakery;
-  };
-
-  /**
-   * Wrapper for making requests via the bakery.
-   *
-   * @private _makeRequest
-   * @param path {String} The JEM endpoint to make the request from,
-   *     e.g. '/env'
-   * @param method {String} The type of http method to use, e.g. GET or POST.
-   * @param params {Object} Optional data object to sent with e.g. POST commands.
-   * @param success {function} A callback to be called on success. Takes
-   *     an xhr object as its only parameter.
-   * @param failure {function} A callback to be called on failure. Takes
-   *     an xhr object as its only parameter.
+     Utility function for making requests via the bakery.
+
+     _makeRequest
+     @param bakery {Object} The bakery object to use.
+     @param path {String} The JEM endpoint to make the request from,
+         e.g. '/env'
+     @param method {String} The type of http method to use, e.g. GET or POST.
+     @param params {Object} Optional data object to sent with e.g. POST commands.
+     @params callback {Function} A callback to handle errors or accept the data
+         from the request. Must accept an error message or null as its first
+         parameter and the response data as its second.
+     @params parse {Boolean} Whether or not to parse the response as JSON.
   */
-  environment.prototype._makeRequest = function(path, method, params, callback) {
+  var _makeRequest = function(bakery, path, method, params, callback, parse) {
```

Comment:
Two things use it, and the method is setup to handle it correctly if nothing is passed in--this cuts back on everything having to set a very optional paramter.

As you can see in other private methods that were lifted from the charmstore, we set the callback in different spots for convenience more than once--since this isn't a method that's exported to anyone to use outside of jujulib, I don't think it's a problem.